### PR TITLE
Add unicode doc to doxygen Tech Notes

### PR DESCRIPTION
**Pull request by byrnHDF**

### CODE REVIEW

File: doxygen/dox/TechnicalNotes.dox [hunk 1]
```diff
@@ -199,6 +200,150 @@ close properly. It will contain data up to the point that it was interrupted.
...
+    lcpl_id =    H5Pcreate(H5P_LINK_CREATE) ;
+    error =      H5Pset_char_encoding(lcpl_id, H5T_CSET_UTF8) ;
+    dataset_id = H5Dcreate2(group_id, "datos_ñ", datatype_id, dataspace_id, 
+                 lcpl_id, H5P_DEFAULT, H5P_DEFAULT) ;               
+\endcode
+
+If the character encoding of an attribute name is unknown, the combination of an
+#H5Aget_create_plist call and an #H5Pget_char_encoding call will reveal that information.
+If the character encoding of an object name is unknown, the information can be accessed
+through the object’sH5L_info_t structure which can be obtained using #H5Lvisit or #H5Lget_info_by_idx calls.
```
Comment (hyoklee) [inline doxygen/dox/TechnicalNotes.dox]: `object’sH5L_info_t` -> `object's H5L_info_t`

**Review by hyoklee — CHANGES_REQUESTED** [review]

File: doxygen/dox/TechnicalNotes.dox [hunk 2]
```diff
@@ -199,6 +200,150 @@ close properly. It will contain data up to the point that it was interrupted.
...
+\li Multi-byte and variable-size character encodings
+\li Limitations of the ASCII character set
+\li Risks associated with the use of the term plain text
+\li Representation of multiple language alphabets or characters in a single document
+
+More specific technical details will only become important if they affect the specifics of
+your application design or implementation.
+
+\section sec_unicode_support How and Where Is UTF-8 Supported in HDF5?
+HDF5 uses characters in object names (which are actually link names, but that’s a story for a
```
Comment (hyoklee) [inline doxygen/dox/TechnicalNotes.dox]: Can we replace `’` with `'`?

File: doxygen/dox/TechnicalNotes.dox [hunk 3]
```diff
@@ -199,6 +200,150 @@ close properly. It will contain data up to the point that it was interrupted.
...
+With Unicode encoding, number of characters is not synonymous with number of bytes. One must
+get used to thinking in terms of number of characters when talking about content, reserving
+number of bytes for discussions of storage size.
+
+When working with Unicode text, one can no longer assume a 1:1 correspondence between the
+number of characters and the data storage requirement. 
+
+\subsection subsec_unicode_warn_sys System Dependencies
+<b>Linux</b>, <b>Unix</b>, and similar systems generally handle UTF-8 encodings in correct and
+predictable ways. There is an apparent consensus in the Linux community that “UTF-8 is just the right way to go.”
```
Comment (hyoklee) [inline doxygen/dox/TechnicalNotes.dox]: Can we replace `“ ”` with plain `""`?

**Review by hyoklee — COMMENTED** [review]
Please minimize the use of 8-bit characters in code documentation.

**Comment by byrnHDF** [comment]
> Please minimize the use of 8-bit characters in code documentation.

These are all conversions from html/md files and it can be difficult to catch everything - appreciate the checks everyone does!

**Review by hyoklee — APPROVED** [review]

**Review by lrknox — APPROVED** [review]

**Pull request merged**